drivers/ethernet/eth_mcux.c

```diff
@@ -1456,7 +1456,8 @@ static int ptp_clock_mcux_adjust(const struct device *dev, int increment)
 
 	ARG_UNUSED(dev);
 
-	if ((increment <= -NSEC_PER_SEC) || (increment >= NSEC_PER_SEC)) {
+	if ((increment <= (int32_t)-NSEC_PER_SEC) ||
```

Comment:
maybe 'if ((increment <= (int32_t)(-NSEC_PER_SEC)) ||
(increment >= (int32_t)NSEC_PER_SEC))`
better? 
nice shot on this issue. Thanks a lot.

BTW, we are refactorring this driver, and this function is rewriten.

Reply:
That's great!  I've updated this PR in the meantime.